---
f_search_pipeline_wp:
  ros__parameters:
    type: system
    parts:
      f_search_pipeline_wp_node
    modes:
      __DEFAULT__:
        f_search_pipeline_wp_node: active.__DEFAULT__
      fd_spiral_high:
        f_search_pipeline_wp_node: active.fd_spiral_high
      fd_spiral_low:
        f_search_pipeline_wp_node: active.fd_spiral_low
      fd_spiral_medium:
        f_search_pipeline_wp_node: active.fd_spiral_medium
      fd_unground:
        f_search_pipeline_wp_node: inactive

f_search_pipeline_wp_node:
  ros__parameters:
    type: node
    modes:
      __DEFAULT__:
        ros__parameters:
          spiral_altitude: 1.0
      fd_spiral_high:
        ros__parameters:
          spiral_altitude: 3.0
      fd_spiral_medium:
        ros__parameters:
          spiral_altitude: 2.0
      fd_spiral_low:
        ros__parameters:
          spiral_altitude: 1.0

f_follow_pipeline_wp:
  ros__parameters:
    type: system
    parts:
      f_follow_pipeline_wp_node
    modes:
      __DEFAULT__:
        f_follow_pipeline_wp_node: inactive
      fd_generate_follow_wp:
        f_follow_pipeline_wp_node: active.__DEFAULT__
      fd_unground:
        f_follow_pipeline_wp_node: inactive

f_follow_pipeline_wp_node:
  ros__parameters:
    type: node
    modes:
      __DEFAULT__:
        ros__parameters:
          random_param: 2.0

f_maintain_motion:
  ros__parameters:
    type: system
    parts:
      f_maintain_motion_node
    modes:
      __DEFAULT__:
        f_maintain_motion_node: active.__DEFAULT__
      fd_set_speed_high:
        f_maintain_motion_node: active.fd_set_speed_high
      fd_set_speed_medium:
        f_maintain_motion_node: active.fd_set_speed_medium
      fd_set_speed_low:
        f_maintain_motion_node: active.fd_set_speed_low
      fd_unground:
        f_maintain_motion_node: inactive

f_maintain_motion_node:
  ros__parameters:
    type: node
    modes:
      __DEFAULT__:
        ros__parameters:
          speed: 2.0
      fd_set_speed_high:
        ros__parameters:
          speed: 3.0
      fd_set_speed_medium:
        ros__parameters:
          speed: 2.0
      fd_set_speed_low:
        ros__parameters:
          speed: 1.0

f_recharge_wp:
  ros__parameters:
    type: system
    parts:
      f_recharge_wp_node
    modes:
      __DEFAULT__:
        f_recharge_wp_node: inactive
      fd_generate_recharge_wp:
        f_recharge_wp_node: active.__DEFAULT__
      fd_unground:
        f_recharge_wp_node: inactive

f_recharge_wp_node:
  ros__parameters:
    type: node
    modes:
      __DEFAULT__:
        ros__parameters:
          random_param: 2.0